pkg/cmd/project/item-edit/item_edit.go

```diff
@@ -108,13 +115,42 @@ func NewCmdEditItem(f *cmdutil.Factory, runF func(config editItemConfig) error)
 	editItemCmd.Flags().StringVar(&opts.date, "date", "", "Date value for the field (YYYY-MM-DD)")
 	editItemCmd.Flags().StringVar(&opts.singleSelectOptionID, "single-select-option-id", "", "ID of the single select option value to set on the field")
 	editItemCmd.Flags().StringVar(&opts.iterationID, "iteration-id", "", "ID of the iteration value to set on the field")
+	editItemCmd.Flags().BoolVar(&opts.clear, "clear", false, "Remove field value")
 
 	_ = editItemCmd.MarkFlagRequired("id")
 
 	return editItemCmd
 }
 
 func runEditItem(config editItemConfig) error {
+	// when clear flag is used, remove value set to the corresponding field ID
+	if config.opts.clear {
+		if config.opts.fieldID == "" && config.opts.projectID == "" {
+			return cmdutil.FlagErrorf("field-id and project-id must be provided for use with the clear flag")
+		}
+		if config.opts.fieldID == "" {
+			return cmdutil.FlagErrorf("field-id must be provided for use with the clear flag")
+		}
```

Comment:
The code here logically makes sense, but I want to reach out to @mntlty being the primary `gh project` maintainer about whether 1) these conditional checks apply to all the variant modes here and 2) whether `runEditItem` should be chopped up into 3 functions for clearing item, updating draft item, updating item values.

Reply:
re: 2 splitting this up into functions makes sense if the end result is more readeable 👍 

re: 1 some of this code looks to be repeated, there might be opportunities for a refactor here. grouping into functions might also help with that. 

noting that there is the mutually exclusive flag above https://github.com/cli/cli/blob/cfe055c29eb846b591f6528083a369c9bb6a4215/pkg/cmd/project/item-edit/item_edit.go#L75 which is preferred to the check below which returns the error `"cannot use this flag in conjunction with clear"`

also noting that `itemID` is a required value according to https://docs.github.com/en/graphql/reference/input-objects#clearprojectv2itemfieldvalueinput but isn't being checked for as a required value here

Reply:
Great points all around! I have used the mutually exclusive function now: https://github.com/cli/cli/pull/8024/commits/6f6e179c49ea728c4925be7fb3bf1f4e9928d250.

Also split the clearing of item value, updating draft item and update item values into separate functions, along with field ID and project ID presence check refactored into a separate function: https://github.com/cli/cli/pull/8024/commits/45e4bf7fc0a791e78e39c62f8133bbaa50b0bf36.

>also noting that itemID is a required value according to https://docs.github.com/en/graphql/reference/input-objects#clearprojectv2itemfieldvalueinput but isn't being checked for as a required value here

@mntlty My apologies if I am missing something straightforward, but could you help me understand what you mean here? Doesn't the required flag check for "ID" already exist? 🤔 

https://github.com/cli/cli/blob/b4d075ef00c44df794c6e7f7465330559289fdc7/pkg/cmd/project/item-edit/item_edit.go#L112

Reply:
> > also noting that itemID is a required value according to https://docs.github.com/en/graphql/reference/input-objects#clearprojectv2itemfieldvalueinput but isn't being checked for as a required value here
> 
> @mntlty My apologies if I am missing something straightforward, but could you help me understand what you mean here? Doesn't the required flag check for "ID" already exist? 🤔
> 

@arunsathiya : That is my understanding as well from reading [`spf13/cobra` code](https://github.com/spf13/cobra/blob/bd4d1655f6655f690f45d281fe698dacb9c7d750/shell_completions.go#L21-L26):

> ```go
> // MarkFlagRequired instructs the various shell completion implementations to
> // prioritize the named flag when performing completion,
> // and causes your command to report an error if invoked without the flag.
> func (c *Command) MarkFlagRequired(name string) error {
> 	return MarkFlagRequired(c.Flags(), name)
> }
> ```